compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolTable.java

```diff
@@ -882,11 +882,11 @@ private void defineNilableIntegerBitwiseOperations(OperatorKind opKind) {
         for (int i = 0; i < unsignedNilableIntTypes.length; i++) {
             for (int j = 0; j < unsignedNilableIntTypes.length; j++) {
                 defineBinaryOperator(opKind, unsignedNilableIntTypes[i], unsignedNilableIntTypes[j],
-                        i <= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
+                        i >= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
                 defineBinaryOperator(opKind, unsignedNilableIntTypes[i], unsignedIntTypes[j],
-                        i <= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
+                        i >= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
                 defineBinaryOperator(opKind, unsignedIntTypes[i], unsignedNilableIntTypes[j],
-                        i <= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
+                        i >= j ? unsignedNilableIntTypes[i] : unsignedNilableIntTypes[j]);
```

Comment:
Don't we have tests that cover nilable operands? Shall we add both negative and positive tests?

Reply:
Only added for bitwise `and`, will add for `or` and `xor`